crates/node/peers/src/store.rs

```diff
@@ -18,21 +18,92 @@ const MAX_PEERS: usize = 2048;
 ///
 /// When the number of peers within the [`BootStore`] exceeds `MAX_PEERS`, the oldest peers are
 /// removed to make room for new ones.
-#[derive(Debug, serde::Serialize)]
+#[derive(Debug, serde::Serialize, Default)]
 pub struct BootStore {
-    /// The file path for the [`BootStore`].
+    /// The file for the [`BootStore`].
     #[serde(skip)]
-    pub path: PathBuf,
+    pub file: Option<File>,
     /// [`Enr`]s for peers.
     pub peers: VecDeque<Enr>,
 }
 
+/// The bootstore caching policy.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum BootStoreFile {
+    /// Default path for the bootstore, ie `~/.kona/<chain_id>/bootstore.json`.
+    Default {
+        /// The l2 chain ID.
+        chain_id: u64,
+    },
+    /// A custom bootstore path is used. This must be a valid path to a file.
+    Custom(PathBuf),
+}
+
+impl TryInto<BootStore> for File {
+    type Error = std::io::Error;
+
+    fn try_into(self) -> Result<BootStore, std::io::Error> {
+        let peers = peers_from_file(&self);
+        Ok(BootStore { file: Some(self), peers })
+    }
+}
+
+impl TryInto<File> for BootStoreFile {
+    type Error = std::io::Error;
+
+    /// Returns a pointer to the bootstore file for the given combination of chain id and bootstore
+    /// file type.
+    fn try_into(self) -> Result<File, std::io::Error> {
+        let path = TryInto::<PathBuf>::try_into(self)?;
+        File::options().read(true).write(true).create(true).truncate(false).open(path)
+    }
+}
+
+impl TryInto<BootStore> for BootStoreFile {
+    type Error = std::io::Error;
+
+    fn try_into(self) -> Result<BootStore, std::io::Error> {
+        let file = TryInto::<File>::try_into(self)?;
+        file.try_into()
+    }
+}
+
+impl TryInto<PathBuf> for BootStoreFile {
+    type Error = std::io::Error;
+
+    fn try_into(self) -> Result<PathBuf, std::io::Error> {
+        match self {
+            Self::Default { chain_id } => {
+                let mut path = dirs::home_dir()
+                    .ok_or(std::io::Error::other("Failed to get home directory"))?;
+                path.push(".kona");
+                path.push(chain_id.to_string());
+                path.push("bootstore.json");
+                Ok(path)
+            }
+            Self::Custom(path) => Ok(path),
+        }
+    }
+}
+
+fn peers_from_file(file: &File) -> VecDeque<Enr> {
+    debug!(target: "bootstore", "Reading boot store from disk: {:?}", file);
+    let reader = BufReader::new(file);
+    match serde_json::from_reader(reader).map(|s: BootStore| s.peers) {
+        Ok(peers) => peers,
+        Err(e) => {
+            warn!(target: "bootstore", "Failed to read boot store from disk: {:?}", e);
+            VecDeque::new()
+        }
+    }
+}
```

Comment:
This feels like it could be an associated method on the `BootStoreFile` so it's nicely exported and accessible without having a private, hidden function.